lib/collection/src/lib.rs

```diff
@@ -447,6 +456,52 @@ impl Collection {
         self.shards.values()
     }
 
+    fn send_shard(&mut self, _shard_id: ShardId) {
+        todo!()
+    }
+
+    pub fn start_shard_transfer(
+        &mut self,
+        shard_id: ShardId,
+        to_peer: PeerId,
+        this_peer: PeerId,
+    ) -> CollectionResult<()> {
+        let shard = self.shard_by_id(shard_id).ok_or_else(|| {
+            CollectionError::service_error("Shard {shard_id} is abscent".to_owned())
+        })?;
+        let from_peer = shard.peer_id(this_peer);
+        self.shard_transfers.insert(
```

Comment:
are all nodes tracking all the current ongoing transfer?

Reply:
Yes, I decided that it is better for all nodes to track progress. This way they can simply apply the transfer when it is finished just by shard id. And also leader might need to know the states in case it needs to interfere (we don't have these cases yet, but I am not sure - we might add them later). As any node can be elected as leader during the transfer - they all need to have the same knowledge.